force_torque_sensor_broadcaster/src/force_torque_sensor_broadcaster.cpp

```diff
@@ -0,0 +1,175 @@
+  }
+
+  frame_id_ = node_->get_parameter("frame_id").as_string();
+  if (frame_id_.empty()) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'frame_id' parameter has to be provided.");
+    return CallbackReturn::ERROR;
+  }
+
+  if (!sensor_name_.empty()) {
+    force_torque_sensor_ = std::make_unique<semantic_components::ForceTorqueSensor>(
+      semantic_components::ForceTorqueSensor(sensor_name_));
+  } else {
+    force_torque_sensor_ = std::make_unique<semantic_components::ForceTorqueSensor>(
+      semantic_components::ForceTorqueSensor(
+        interface_names_[0], interface_names_[1],
+        interface_names_[2], interface_names_[3], interface_names_[4], interface_names_[5]));
+  }
+
+  try {
+    // register ft sensor data publisher
+    sensor_state_publisher_ = node_->create_publisher<geometry_msgs::msg::WrenchStamped>(
+      "~/sensor_states", rclcpp::SystemDefaultsQoS());
+    realtime_publisher_ = std::make_unique<StatePublisher>(sensor_state_publisher_);
+  } catch (...) {
+    return CallbackReturn::ERROR;
+  }
+
+  realtime_publisher_->lock();
+  realtime_publisher_->msg_.header.frame_id = frame_id_;
+  realtime_publisher_->unlock();
+
+  RCLCPP_INFO_STREAM(get_node()->get_logger(), "configure successful");
+  return CallbackReturn::SUCCESS;
+}
+
+controller_interface::InterfaceConfiguration
+ForceTorqueSensorBroadcaster::command_interface_configuration() const
+{
+  controller_interface::InterfaceConfiguration command_interfaces_config;
+  command_interfaces_config.type = controller_interface::interface_configuration_type::NONE;
+
+  return command_interfaces_config;
+}
+
+controller_interface::InterfaceConfiguration
+ForceTorqueSensorBroadcaster::state_interface_configuration() const
+{
+  controller_interface::InterfaceConfiguration state_interfaces_config;
+  state_interfaces_config.type = controller_interface::interface_configuration_type::INDIVIDUAL;
+
+  state_interfaces_config.names = force_torque_sensor_->get_state_interface_types();
+
+  return state_interfaces_config;
+}
+
+CallbackReturn ForceTorqueSensorBroadcaster::on_activate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  RCLCPP_INFO(node_->get_logger(), "Number of state interface is: %d.", state_interfaces_.size());
+  force_torque_sensor_->assign_loaned_state_interfaces(state_interfaces_);
+  return CallbackReturn::SUCCESS;
+}
+
+CallbackReturn ForceTorqueSensorBroadcaster::on_deactivate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  force_torque_sensor_->release_interfaces();
+  return CallbackReturn::SUCCESS;
+}
+
+controller_interface::return_type ForceTorqueSensorBroadcaster::update()
+{
+  if (realtime_publisher_ && realtime_publisher_->trylock()) {
+    realtime_publisher_->msg_.header.stamp = node_->now();
+    RCLCPP_INFO(node_->get_logger(), "Trying to get values");
+    realtime_publisher_->msg_.wrench = force_torque_sensor_->get_values_as_message();
+    RCLCPP_INFO(node_->get_logger(), "Aftert getting values");
```

Comment:
```suggestion
    RCLCPP_DEBUG(node_->get_logger(), "Aftert getting values");
```

Reply:
Changed to,
`RCLCPP_DEBUG(node_->get_logger(), "After getting values");`